cmd/add_subnet_validator.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/subnet-cli/pkg/color"
+	"github.com/manifoldco/promptui"
+	"github.com/onsi/ginkgo/v2/formatter"
+	"github.com/spf13/cobra"
+)
+
+func newAddSubnetValidatorCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "subnet-validator",
+		Short: "Adds a subnet to the validator",
+		Long: `
+Adds a subnet to the validator.
+
+$ subnet-cli add subnet-validator \
+--private-key-path=.insecure.ewoq.key \
+--public-uri=http://localhost:52250 \
+--subnet-id="24tZhrm8j8GCJRE9PomW8FaeqbgGS4UAQjJnqqn8pq5NwYSYV1" \
+--node-id="NodeID-4B4rc5vdD1758JSBYL1xyvE5NHGzz6xzH" \
+--validate-weight=1000
+
+`,
+		RunE: createSubnetValidatorFunc,
+	}
+
+	cmd.PersistentFlags().StringVar(&subnetIDs, "subnet-id", "", "subnet ID (must be formatted in ids.ID)")
+	cmd.PersistentFlags().StringVar(&nodeIDs, "node-id", "", "node ID (must be formatted in ids.ID)")
+
+	start := time.Now().Add(time.Minute)
+	end := start.Add(50 * 24 * time.Hour)
+	cmd.PersistentFlags().StringVar(&validateStarts, "validate-start", start.Format(time.RFC3339), "validate start timestamp in RFC3339 format")
+	cmd.PersistentFlags().StringVar(&validateEnds, "validate-end", end.Format(time.RFC3339), "validate start timestamp in RFC3339 format")
+	cmd.PersistentFlags().Uint64Var(&validateWeight, "validate-weight", 1000, "validate weight")
+
+	return cmd
+}
+
+var errZeroValidateWeight = errors.New("zero validate weight")
+
+func createSubnetValidatorFunc(cmd *cobra.Command, args []string) error {
+	cli, info, err := InitClient(publicURI, true)
+	if err != nil {
+		return err
+	}
+	info.txFee = uint64(info.feeData.TxFee)
+	info.nodeID, err = ids.ShortFromPrefixedString(nodeIDs, constants.NodeIDPrefix)
+	if err != nil {
+		return err
+	}
+	info.subnetID, err = ids.FromString(subnetIDs)
+	if err != nil {
+		return err
+	}
+	info.validateStart, err = time.Parse(time.RFC3339, validateStarts)
+	if err != nil {
+		return err
+	}
+	info.validateEnd, err = time.Parse(time.RFC3339, validateEnds)
```

Comment:
We should make sure this is <= staker validation end and that the NodeID is a staker on the primary network